Dockerfile

```diff
@@ -1,49 +1,59 @@
-FROM node:10.3-alpine
-LABEL maintainer="https://github.com/verdaccio/verdaccio"
+FROM node:10.3-alpine as builder
 
-RUN apk --no-cache add openssl && \
-    wget -O /usr/local/bin/dumb-init https://github.com/Yelp/dumb-init/releases/download/v1.2.0/dumb-init_1.2.0_amd64 && \
-    chmod +x /usr/local/bin/dumb-init && \
-    apk del openssl && \
-    apk --no-cache add ca-certificates wget && \
+RUN apk --no-cache add openssl ca-certificates wget && \
     wget -q -O /etc/apk/keys/sgerrand.rsa.pub https://raw.githubusercontent.com/sgerrand/alpine-pkg-glibc/master/sgerrand.rsa.pub && \
-    wget https://github.com/sgerrand/alpine-pkg-glibc/releases/download/2.25-r0/glibc-2.25-r0.apk && \
+    wget -q https://github.com/sgerrand/alpine-pkg-glibc/releases/download/2.25-r0/glibc-2.25-r0.apk && \
     apk add glibc-2.25-r0.apk
 
-ENV APPDIR /usr/local/app
-
-WORKDIR $APPDIR
-
-ADD . $APPDIR
+WORKDIR /opt/verdaccio-build
+COPY . .
 
-ENV NODE_ENV=production
+ENV NODE_ENV=production \
+    VERDACCIO_BUILD_REGISTRY=https://registry.npmjs.org/
 
-RUN npm config set registry http://registry.npmjs.org/ && \
-    yarn global add -s flow-bin@0.69.0 && \
+RUN yarn config set registry $VERDACCIO_BUILD_REGISTRY && \
     yarn install --production=false && \
     yarn lint && \
     yarn code:docker-build && \
     yarn build:webui && \
     yarn cache clean && \
     yarn install --production=true --pure-lockfile
 
+
+
+FROM node:10.3-alpine
+LABEL maintainer="https://github.com/verdaccio/verdaccio"
+
+RUN apk --no-cache add openssl dumb-init
+
 RUN mkdir -p /verdaccio/storage /verdaccio/plugins /verdaccio/conf
 
+ENV VERDACCIO_APPDIR=/opt/verdaccio
+WORKDIR $VERDACCIO_APPDIR
+
+COPY --from=builder /opt/verdaccio-build .
+
 ADD conf/docker.yaml /verdaccio/conf/config.yaml
 
-RUN addgroup -S verdaccio && adduser -S -G verdaccio verdaccio && \
-    chown -R verdaccio:verdaccio "$APPDIR" && \
-    chown -R verdaccio:verdaccio /verdaccio
+ENV PATH=${VERDACCIO_APPDIR}/bin:${PATH} \
```

Comment:
Can we unify in a unique ENV sentence after LABEL?

Reply:
Updated.

Reply:
Sorry, nope, I have at least to break it into two env sentences, I'm not allowed to reference an `ENV` definition before the sentence is finished. So the PATH has to be updated afterwards.